chain-impl-mockchain/src/testing/arbitrary/address.rs

```diff
@@ -95,31 +95,24 @@ impl ArbitraryAddressDataValueVec {
         self.0
             .iter()
             .cloned()
-            .filter(|x| match x.address_data.kind() {
-                Kind::Single { .. } => true,
-                _ => false,
-            })
+            .filter(|x| matches!(x.address_data.kind(), Kind::Single { .. }))
             .collect()
     }
+    #[allow(clippy::match_like_matches_macro)]
```

Comment:
These are redundant because you have changed the expressions to use `matches!`. I have removed these overrides in another commit.